packages/koa-shopify-auth/src/auth/create-enable-cookies.ts

```diff
@@ -0,0 +1,425 @@
+import {Context} from 'koa';
+
+import Error from './errors';
+import {OAuthStartOptions} from "../types";
+
+export default function createEnableCookies({apiKey}: OAuthStartOptions) {
+  return function enableCookies(ctx: Context) {
+    const HEADING = 'Enable cookies';
+    const BODY = 'You must manually enable cookies in this browser in order to use this app within Shopify.'
+    const FOOTER = `Cookies let the app authenticate you by temporarily storing your preferences and personal
+information. They expire after 30 days.`
+    const ACTION = 'Enable cookies';
+
+    const {query} = ctx;
+    const {shop} = query;
+
+    if (shop == null) {
+      ctx.throw(400, Error.ShopParamMissing);
+      return;
+    }
+
+    ctx.body = `
```

Comment:
Is there a better way/place to do/put this HTML?

Reply:
There's no better way to do this than a template literal if we don't want to depend on some kind of templating library (or ejs), and need interpolation. Similar to the routing being a bunch of dumb ifs we opted for minimizing the dependencies in this package since it's relatively simple.

If it was just a static page we could have koa just serve the file directly or something.

Reply:
I'm not sure there's a good way to share the templates that would be consumable by both projects and not unnecessarily complex.